src/components/navigation/Navbar.tsx

```diff
@@ -25,17 +24,39 @@ interface NavLinksProps {
 
 interface NavbarProps extends BoxProps, React.PropsWithChildren {
   logo?: ImageColorSchemeSwitchType | "theme" | null;
+  linkComponent?: React.ElementType;
+}
+
+interface NavLinkProps extends LinkProps {
+  children: React.ReactNode;
+  linkComponent?: React.ElementType;
+  to?: string;
+  href?: string;
 }
 
-const NavLink = ({ children, ...props }: LinkProps) => {
+const NavLink = ({
+  children,
+  linkComponent,
+  to,
+  href,
+  ...props
+}: NavLinkProps) => {
   const theme = useTheme();
 
+  const shouldUseLinkComponent = linkComponent && to;
+
+  const linkProps = shouldUseLinkComponent
```

Comment:
Don't think we need this one either. (But we do still need it on linkComponent on NavbarProps)

Reply:
Leave it.